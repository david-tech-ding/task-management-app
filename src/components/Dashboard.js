import React, { useState, useEffect } from "react";
import axios from "axios";
import TaskSummary from "./TaskSummary";
import { Link } from "react-router-dom";
import "../styles/dashboard.css";

const Dashboard = ({ user }) => {
  axios.defaults.baseURL = "http://localhost:3001";
  const [tasks, setTasks] = useState([]);

  useEffect(() => {
    axios
      .get("/task")
      .then((data) => {
        setTasks(data.data);
      })
      .catch((err) => console.log(err));
  }, []);

  const yourTasks = tasks.filter((task) => task.assignTo === user);

  const tasksAssignedByYou = tasks.filter((task) => task.assignedBy === user);

  const soonDue = new Date();
  soonDue.setDate(new Date().getDate() + 3);
  console.log(soonDue);

  const yourTasksDueSoon = yourTasks.filter((yourTask) => {
    const taskDueDate = new Date(yourTask.dueDate);
    return taskDueDate <= soonDue;
  });

  return (
    <div className="dashboard-page">
      <h1>{user}'s Dashboard</h1>
      <TaskSummary
        title={
          <Link className="task-summary-link" to="/tasks">
            Your Tasks
          </Link>
        }
        tasks={yourTasks}
      />
      <TaskSummary title="Tasks Assigned By You" tasks={tasksAssignedByYou} />
      <TaskSummary title="Due Soon" tasks={yourTasksDueSoon} />
    </div>
  );
};

export default Dashboard;
