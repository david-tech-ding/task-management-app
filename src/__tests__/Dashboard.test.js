import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter as Router } from "react-router-dom";
import Dashboard from "../components/Dashboard";

describe("Dashboard component", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <Router>
        <Dashboard user={[]} />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
