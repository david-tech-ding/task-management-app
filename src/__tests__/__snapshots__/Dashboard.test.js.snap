// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard component renders correctly 1`] = `
<DocumentFragment>
  <div
    class="dashboard-page"
  >
    <h1>
      's Dashboard
    </h1>
    <div
      class="dashboard"
    >
      <div
        class="task-summary"
      >
        <h2
          class="task-summary-heading"
        >
          <a
            class="task-summary-link"
            href="/your-tasks"
          >
            Your Tasks
          </a>
        </h2>
        <div
          class="high-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            High Priority
          </h3>
          0
        </div>
        <div
          class="medium-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            Medium Priority
          </h3>
          0
        </div>
        <div
          class="low-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            Low Priority
          </h3>
          0
        </div>
      </div>
    </div>
    <div
      class="dashboard"
    >
      <div
        class="task-summary"
      >
        <h2
          class="task-summary-heading"
        >
          <a
            class="task-summary-link"
            href="/assigned-by-you"
          >
            Assigned By You
          </a>
        </h2>
        <div
          class="high-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            High Priority
          </h3>
          0
        </div>
        <div
          class="medium-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            Medium Priority
          </h3>
          0
        </div>
        <div
          class="low-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            Low Priority
          </h3>
          0
        </div>
      </div>
    </div>
    <div
      class="dashboard"
    >
      <div
        class="task-summary"
      >
        <h2
          class="task-summary-heading"
        >
          <a
            class="task-summary-link"
            href="/due-soon"
          >
            Due Soon
          </a>
        </h2>
        <div
          class="high-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            High Priority
          </h3>
          0
        </div>
        <div
          class="medium-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            Medium Priority
          </h3>
          0
        </div>
        <div
          class="low-priority-summary"
        >
          <h3
            class="priority-summary-heading"
          >
            Low Priority
          </h3>
          0
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
